WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelPackagesViewModel.kt

```
@@ -0,0 +1,102 @@
+import com.woocommerce.android.ui.products.ParameterRepository
+import com.woocommerce.android.ui.products.ProductDetailRepository
+import com.woocommerce.android.ui.products.models.SiteParameters
+import com.woocommerce.android.util.CoroutineDispatchers
+import com.woocommerce.android.viewmodel.LiveDataDelegate
+import com.woocommerce.android.viewmodel.SavedStateWithArgs
+import com.woocommerce.android.viewmodel.ScopedViewModel
+import kotlinx.android.parcel.Parcelize
+import kotlinx.coroutines.launch
+
+class EditShippingLabelPackagesViewModel @AssistedInject constructor(
+    @Assisted savedState: SavedStateWithArgs,
+    dispatchers: CoroutineDispatchers,
+    parameterRepository: ParameterRepository,
+    private val orderDetailRepository: OrderDetailRepository,
+    private val productDetailRepository: ProductDetailRepository
+) : ScopedViewModel(savedState, dispatchers) {
+    companion object {
+        private const val KEY_PARAMETERS = "ket_parameters"
+    }
+
+    private val arguments: EditShippingLabelPackagesFragmentArgs by savedState.navArgs()
+
+    val viewStateData = LiveDataDelegate(savedState, ViewState())
+    private var viewState by viewStateData
+
+    val parameters: SiteParameters by lazy {
+        parameterRepository.getParameters(KEY_PARAMETERS, savedState)
+    }
+
+    init {
+        initState()
+    }
+
+    private fun initState() {
+        if (viewState.shippingLabelPackages.isNotEmpty()) return
+        launch {
+            loadProductsWeightsIfNeeded()
+            val packagesList = arguments.shippingLabelPackages.toList().ifEmpty {
+                val order = requireNotNull(orderDetailRepository.getOrder(arguments.orderId))
+                listOf(
+                    ShippingLabelPackage(
+                        selectedPackage = arguments.availablePackages.first(),
+                        weight = Double.NaN,
+                        items = order.items.map { it.toShippingItem() }
```

This shows all products for the order, including virtual and refunded products, which should be filtered out.

---

I added filtering in the last commit, I'm still ignoring errors if the products API call fails, I prefer to keep it this way in this PR, and confirm with Garance later if we should display an error message, and implement it in a separate PR.